tools/edge_cut/src/surf/BdryTrimmer.inc

```diff
@@ -0,0 +1,241 @@
+typename edge_cut::surf::BdryTrimmer<Polyhedron>::Vertex_const
+edge_cut::surf::BdryTrimmer<Polyhedron>::getPrevTopoVert( Vertex_const i_vertTopo )
+{
+  auto l_edge = i_vertTopo->vertex_begin();
+  while( !l_edge->is_border_edge() ){ l_edge++; }
+
+  return l_edge->prev()->vertex();
+}
+
+template< class Polyhedron >
+typename edge_cut::surf::BdryTrimmer<Polyhedron>::Vertex_const
+edge_cut::surf::BdryTrimmer<Polyhedron>::getNextTopoVert( Vertex_const i_vertTopo )
+{
+  auto l_edge = i_vertTopo->vertex_begin();
+  while( ! l_edge->next()->is_border_edge() ){ l_edge++; }
+
+  return l_edge->next()->vertex();
+}
+
+template< class Polyhedron >
+typename edge_cut::surf::BdryTrimmer<Polyhedron>::Vertex
+edge_cut::surf::BdryTrimmer<Polyhedron>::getBdryVertex( Vertex_const i_v )
+{
+  VertexIt l_vIt = m_bdry.vertices_begin();
+  for ( ;
+        l_vIt != m_bdry.vertices_end();
+        l_vIt++ )
+  {
+    if ( l_vIt->point() == i_v->point() ) break;
+  }
+
+  if ( l_vIt == m_bdry.vertices_end() )
+  {
+    EDGE_LOG_INFO << "Could not find coincident vertex between topography and border mesh - mesh will not be trimmed";
+    EDGE_LOG_INFO << "Missing point: " << i_v->point();
+    assert( l_vIt != m_bdry.vertices_end() );
+  }
+
+  return l_vIt;
+}
+
+template< class Polyhedron >
+typename edge_cut::surf::BdryTrimmer<Polyhedron>::Halfedge
+edge_cut::surf::BdryTrimmer<Polyhedron>::getBdryHalfedge( Vertex_const i_v1, Vertex_const i_v2 )
+{
+  Vertex l_vert1 = getBdryVertex( i_v1 );
+  Vertex l_vert2 = getBdryVertex( i_v2 );
+
+  auto l_edge = l_vert2->vertex_begin();
+  while( l_edge->prev()->vertex()->point() != l_vert1->point() ) { l_edge++; }
+
+  return l_edge;
+}
+
+// Checks the relative orientation of the two meshes. Given a halfedge on the
+// boundary mesh which also represents an edge on the topography mesh,
+// determines whether that halfedge belongs to the face above the topography or
+// below.
+template< class Polyhedron >
+bool
+edge_cut::surf::BdryTrimmer<Polyhedron>::isPosOriented( Halfedge l_halfedge )
+{
+
+  if ( l_halfedge->is_border() ) return true;
+
+  Point v1 = l_halfedge->prev()->vertex()->point();
+  Point v2 = l_halfedge->vertex()->point();
+  Point v3 = l_halfedge->next()->vertex()->point();
+
+  //                   (v2) .-------------------
+  //                      ./
+  // [above topo]       . /
+  //                  .  /
+  //                .   /
+  //         (v3) .    / U      [below topo]
+  //             \    /
+  //            W \  /
+  // ______________\/   (v1)
+  //
+  // Consider vector u = v_2 - v_1   and   w = v_3 - v_1
+  // We will compute the perpendicular vector from u into v_3 : if the z
+  // component of this vector is positive, then we are "above" the topography.
+  // Otherwise, we are below, and have to flip the halfedge to the other side.
+  double l_dotProdUW = (v3.x()-v1.x()) * (v2.x()-v1.x()) + (v3.y()-v1.y()) * (v2.y()-v1.y()) + (v3.z()-v1.z()) * (v2.z()-v1.z());
```

Comment:
Is precision a template in CGAL. In that case, use the respective spec over ``double``?